server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -187,6 +208,17 @@ public static boolean isRemoteStoreClusterStateEnabled(Settings settings) {
             && isRemoteClusterStateAttributePresent(settings);
     }
 
+    private static boolean isRemoteRoutingTableAttributePresent(Settings settings) {
+        return settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY)
+            .isEmpty() == false;
+    }
+
+    public static boolean isRemoteRoutingTableEnabled(Settings settings) {
+        return FeatureFlags.isEnabled(REMOTE_PUBLICATION_EXPERIMENTAL)
+            && RemoteRoutingTableService.REMOTE_ROUTING_TABLE_ENABLED_SETTING.get(settings)
+            && isRemoteRoutingTableAttributePresent(settings);
```

Comment:
I understand the check on feature flag but if feature flag is enabled, but do we need check both cluster settings and presence of remote routing table attributes? Ideally, if cluster setting is enabled, then routing table attributes must be present.

Reply:
Ideally attributes should be present, but just keeping this as an extra check since the feature will fail if for some reason repo is not there. This is what we do for remote state as well.

Reply:
> since the feature will fail if for some reason repo is not there.

Agree. IMO, node bootstrap should fail if cluster setting is enabled and repo details are not provided.